launchflow/aws/api_gateway.py

```diff
@@ -60,4 +66,158 @@ def inputs(self, environment_state: EnvironmentState) -> APIGatewayInputs:
+        "my-api-gateway-route",
+        api_gateway=api_gateway,
+        function=function,
+    )
+    ```
+    """
+
+    product = ResourceProduct.AWS_API_GATEWAY_LAMBDA_INTEGRATION.value
+
+    def __init__(
+        self,
+        name: str,
+        *,
+        api_gateway: APIGateway,
+        # NOTE: This only supports Lambda function integrations for now
+        function: LambdaFunction,
+    ) -> None:
+        """TODO"""
+        super().__init__(
+            name=name,
+            resource_id=f"{name}-{lf.project}-{lf.environment}",
+        )
+        self._api_gateway = api_gateway
+        self._function = function
+        self.depends_on(
+            api_gateway, function
+        )  # TODO: Remove this once we fix lf.node.Depends
+
+    def inputs(
+        self, environment_state: EnvironmentState
+    ) -> APIGatewayLambdaIntegrationInputs:
+        """Get the inputs required for the API Gateway Lambda Integration.
+
+        **Args:**
+         - `environment_state (EnvironmentState)`: The environment to get inputs for
+
+        **Returns:**
+         - `APIGatewayLambdaIntegrationInputs`: The inputs required for the API Gateway Lambda Integration
+        """
+
+        return APIGatewayLambdaIntegrationInputs(
+            resource_id=self.resource_id,
+            api_gateway_id=Depends(self._api_gateway).api_gateway_id,  # type: ignore
+            function_arn=Depends(self._function).aws_arn,  # type: ignore
+            function_alias=Depends(self._function).alias_name,  # type: ignore
+        )
+
+
+# TODO: Expose more options for APIGatewayRoute
+@dataclass
+class APIGatewayRouteInputs(ResourceInputs):
+    api_gateway_id: str
+    route_key: str
+    authorization: Literal["NONE", "AWS_IAM", "JWT", "CUSTOM"]
+    api_integration_id: Optional[str]
+
+
+@dataclass
+class APIGatewayRouteOutputs(Outputs):
+    api_route_id: str
+
+
+class APIGatewayRoute(AWSResource[APIGatewayRouteOutputs]):
+    """An API Gateway Route
+
+    ****Example usage:****
+    ```python
+    import launchflow as lf
+
+    api_gateway = lf.aws.APIGateway("my-api-gateway")
+    route = lf.aws.APIGatewayRoute("my-api-gateway-route", api_gateway=api_gateway)
+    ```
+    """
+
+    product = ResourceProduct.AWS_API_GATEWAY_ROUTE.value
+
+    def __init__(
+        self,
+        name: str,
+        *,
+        api_gateway: APIGateway,
+        route_key: str = "$default",
+        # NOTE: We don't support JWT or CUSTOM authorization yet
+        authorization: Literal["NONE", "AWS_IAM"] = "NONE",  # NONE == public
+        api_gateway_integration: Optional[APIGatewayLambdaIntegration] = None,
+    ) -> None:
+        """TODO"""
+        super().__init__(
+            name=name,
+            resource_id=f"{name}-{lf.project}-{lf.environment}",
+        )
+        self._api_gateway = api_gateway
+        self.depends_on(api_gateway)  # TODO: Remove this once we fix lf.node.Depends
```

Comment:
what's wrong with depends I thought it worked?

Reply:
not 100% sure why, but there's cases where we dont include the depends in the flowstate without this